blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -81,14 +81,17 @@ public static void send(Message message, OutputStream outputStream) throws IOExc
             toSend.append("\n");
         }
         toSend.append("\n");
-        outputStream.write(toSend.toString().getBytes());
+        synchronized (outputStream) {
+            outputStream.write(toSend.toString().getBytes());
 
-        if (message.getBody() != null) {
-            outputStream.write(message.getBody());
-        }
+            if (message.getBody() != null) {
+                outputStream.write(message.getBody());
+            }
 
-        outputStream.write("\000\n\n".getBytes());
+            outputStream.write("\000\n\n".getBytes());
```

Comment:
If we are sending "\000\n\n" a lot of times it may be beneficial to pull it out as a constant:

    static final byte[] eom =  "\000\n\n".getBytes()